closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -367,6 +373,54 @@ private static void checkFIPSCompatibility(Properties props) {
         }
     }
 
+    /**
+     * Check whether a security property can be set.
+     *
+     * A security property that is set by a RestrictedSecurity profile,
+     * while FIPS security mode is enabled, cannot be reset programmatically.
+     *
+     * Every time an attempt to set a security property is made, a check is
+     * performed. If the above scenario holds true, a SecurityException is
+     * thrown.
+     *
+     * One can override this behaviour and allow the user to set any security
+     * property through the use of {@code -Dsemeru.fips.allowsetproperties=true}.
+     *
+     * @param key the security property that the user wants to set
+     * @throws SecurityException
+     *         if the security property is set by the profile and cannot
+     *         be altered
+     */
+    public static void checkSetSecurityProperty(String key) {
+        if (debug != null) {
+            debug.println("RestrictedSecurity: Checking whether security property '"
+                    + key + "' can be set.");
+        }
+
+        /*
+         * Only disallow setting of security properties that are set by the active profile,
+         * if FIPS has been enabled.
+         *
+         * Allow any change, if the 'semeru.fips.allowsetproperties' flag is set to true.
+         */
+        if (userEnabledFIPS && !allowSetSecurityProps && unmodifiableProperties.contains(key)) {
```

Comment:
Only the final part of this test is needed:
```java
        if (unmodifiableProperties.contains(key)) {
```

Reply:
Right. Cause we checked when adding to the set. Updated.